### PR TITLE
BREAKING CHANGE: remove old amountMath

### PR DESCRIPTION
**Pull request by katelynsills**
Merging is blocked by removing the old amountMath in the dapps (https://github.com/Agoric/agoric-sdk/issues/2603), but we can still get it reviewed.

### CODE REVIEW

**Comment by katelynsills** [comment]
@erights, I think the majority of the review is for you but @michaelfig could you check on the wallet and any other code you're the owner for? And @FUDCo I made some changes to swingset-runner tests. Thanks!

**Review by FUDCo — APPROVED** [review]
The changes to the swingset-runner tests look fine.  Overall, very tidy for something that touches so many files.  The cleanerness (is that a word?) of the new API really comes through when you look at all the diffs.

**Comment by katelynsills** [comment]
Adding @Chris-Hibbert to take over part or all of @erights's review. @Chris-Hibbert, can you check ERTP and Zoe especially? Thanks!

**Review by Chris-Hibbert — APPROVED** [review]
one tiny cleanup, and one oversight.  Looks good!

File: packages/deploy-script-support/src/saveIssuer.js [hunk 1]
```diff
@@ -18,9 +14,8 @@ export const makeSaveIssuer = (
       brandPetname,
       pursePetname,
     );
-    const localAmountMathSavedP = saveLocalAmountMaths([brandPetname]);
 
-    const result = await Promise.all([emptyPurseMadeP, localAmountMathSavedP]);
+    const result = await emptyPurseMadeP;
```
[the remark this answers is not in the document]
Comment (katelynsills) [reply packages/deploy-script-support/src/saveIssuer.js]: Yup, I think that's right

File: packages/deploy-script-support/test/unitTests/test-offer.js [hunk 2]
```diff
@@ -55,36 +55,24 @@ test('offer', async t => {
...
-    throw Error('not found');
-  };
-
   const { offer } = makeOfferAndFindInvitationAmount(
     walletAdmin,
     zoe,
     zoeInvitationPurse,
-    getLocalAmountMath,
-    invitationMath,
+    invitationBrand,
```
[the remark this answers is not in the document]
Comment (katelynsills) [reply packages/deploy-script-support/test/unitTests/test-offer.js]: good catch!

**Review by dtribble — COMMENTED** [review]
WIP 21/50 reviewed

File: packages/cosmic-swingset/lib/ag-solo/vats/bootstrap.js [hunk 3]
```diff
@@ -251,7 +251,7 @@ export function buildRootObject(vatPowers, vatParameters) {
             },
             want: {
               // We just throw away our governance tokens.
-              Governance: amountMath.make(0n, govBrand),
+              Governance: amountMath.make(govBrand, 0n),
```
[the remark this answers is not in the document]
Comment (katelynsills) [reply packages/cosmic-swingset/lib/ag-solo/vats/bootstrap.js]: That would be an improvement, but I was trying not to make any improvements that aren't required for switching to the new amountMath. I'll make this improvement and any other ones that you point out, but I want to keep this PR in scope :)

File: packages/dapp-svelte-wallet/api/test/test-lib-wallet.js [hunk 4]
```diff
@@ -166,7 +166,7 @@ test('lib-wallet issuer and purse methods', async t => {
   const moolaPurse = wallet.getPurse('fun money');
   t.deepEqual(
     await moolaPurse.getCurrentAmount(),
-    moolaBundle.amountMath.getEmpty(),
+    amountMath.makeEmpty(moolaBundle.brand, MathKind.NAT),
```
[the remark this answers is not in the document]
Comment (katelynsills) [reply packages/dapp-svelte-wallet/api/test/test-lib-wallet.js:169]: I think that's out of scope for this PR, and also it's a test so refactoring is probably lower priority.

File: packages/deploy-script-support/test/unitTests/test-depositInvitation.js [hunk 5]
```diff
@@ -1,19 +1,16 @@
...
 
 test('depositInvitation', async t => {
-  const { mint, issuer, amountMath } = makeIssuerKit(
-    'invitations',
-    MathKind.SET,
-  );
+  const { mint, issuer, brand } = makeIssuerKit('invitations', MathKind.SET);
   const purse = issuer.makeEmptyPurse();
-  const paymentAmount = amountMath.make(harden([{ instance: {} }]));
+  const paymentAmount = amountMath.make(brand, [{ instance: {} }]);
```
[the remark this answers is not in the document]
Comment (katelynsills) [reply packages/deploy-script-support/test/unitTests/test-depositInvitation.js:13]: nope, not empty.

File: packages/deploy-script-support/test/unitTests/test-findInvitationAmount.js [hunk 6]
```diff
@@ -1,41 +1,36 @@
...
-    getLocalAmountMath,
-    invitationMath,
+    brand,
   );
 
   const notFoundResult = await findInvitationAmount({
     description: 'not found',
   });
-  t.deepEqual(notFoundResult, invitationMath.make(harden([])));
+  t.deepEqual(notFoundResult, amountMath.make(brand, []));
```
[the remark this answers is not in the document]
Comment (katelynsills) [reply packages/deploy-script-support/test/unitTests/test-findInvitationAmount.js]: Yup, this is empty

[the remark this answers is not in the document]
Comment (katelynsills) [reply packages/deploy-script-support/test/unitTests/test-findInvitationAmount.js]: changed.

File: packages/cosmic-swingset/lib/ag-solo/vats/bootstrap.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (katelynsills) [reply packages/cosmic-swingset/lib/ag-solo/vats/bootstrap.js]: changed.

**Pull request merged**